src/libraries/Common/src/Interop/Unix/System.Native/Interop.DeviceFiles.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    // mknod: https://man7.org/linux/man-pages/man2/mknod.2.html
+    // makedev, major and minor: https://man7.org/linux/man-pages/man3/makedev.3.html
+    internal static partial class Sys
+    {
+        internal static int CreateBlockDevice(string pathName, int mode, uint major, uint minor)
+        {
+            return MkNod(pathName, mode | FileTypes.S_IFBLK, major, minor);
```

Comment:
note: regular users don't have permissions to create nodes.

Reply:
Correct, I need to add that to the documentation. If the user wants device files to be included, they would have to elevate the process that is attempting to extract the tar. This would match the behavior of the `tar` tool: If you attempt to extract a device file without elevating, you get the error:

```
tar: myblock: Cannot mknod: Operation not permitted
tar: Exiting with failure status due to previous errors
```

This happens regardless of the uid/gid with which the device file entry was stored in the tar.

Reply:
And I should also add a note at the top of this interop method saying it can only work if elevated.

Reply:
Do you know if being added to a particular group would give you permission to create nodes? Or is it only allowed if you're elevated?

Reply:
Definitely not for this PR, but do you think we should add logic that detects the availability of `CAP_MKNOD` in the current process/thread?

Reply:
We don't need to do anything specific. If the user doesn't have the capability we'll throw `IOException` for `EPERM`, and that seems appropriate.